util/src/lib.rs

```diff
@@ -26,6 +31,7 @@ pub fn log2_ceil(n: usize) -> usize {
 pub fn log2_strict(n: usize) -> usize {
     let res = n.trailing_zeros();
     assert!(n.wrapping_shr(res) == 1, "Not a power of two: {}", n);
+    assume(n == 1 << res);
```

Comment:
It's not so clear what this does, maybe add a comment.

Reply:
Ah, that tells the optimizer about the semantics of `log2_strict`. This way it knows that it can replace `n` with `1 << res` and vice versa. I'll add a comment!